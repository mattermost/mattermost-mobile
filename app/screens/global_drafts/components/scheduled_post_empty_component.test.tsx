// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';

import {renderWithIntl} from '@test/intl-test-helper';

import ScheduledPostEmptyComponent from './scheduled_post_empty_component';

describe('screens/global_drafts/components/ScheduledPostEmptyComponent', () => {
    it('renders all required elements', () => {
        const {getByTestId, getByText} = renderWithIntl(
            <ScheduledPostEmptyComponent/>,
        );

        // Verify component renders
        expect(getByTestId('scheduled_post_empty_component')).toBeTruthy();

        // Verify text elements
        expect(getByText('No scheduled drafts at the moment')).toBeTruthy();
        expect(getByText('Schedule drafts to send messages at a later time. Any scheduled drafts will show up here and can be modified after being scheduled.')).toBeTruthy();
    });
});
