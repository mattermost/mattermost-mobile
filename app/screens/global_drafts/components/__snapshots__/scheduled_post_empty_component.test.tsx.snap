// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/global_drafts/components/ScheduledPostEmptyComponent should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="scheduled_post_empty_component"
>
  <ViewManagerAdapter_ExpoImage
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={
      [
        {
          "testUri": "../../../dist/assets/images/Scheduled_Message.png",
        },
      ]
    }
    style={{}}
    transition={null}
  />
  <Text
    style={
      {
        "color": "#3f4350",
        "fontFamily": "Metropolis-SemiBold",
        "fontSize": 20,
        "fontWeight": "600",
        "lineHeight": 28,
      }
    }
  >
    No scheduled drafts at the moment
  </Text>
  <Text
    style={
      {
        "color": "rgba(63,67,80,0.72)",
        "fontFamily": "OpenSans",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
        "marginTop": 8,
        "textAlign": "center",
      }
    }
  >
    Schedule drafts to send messages at a later time. Any scheduled drafts will show up here and can be modified after being scheduled.
  </Text>
</View>
`;
